webapp/src/components/CreateEstatePage/CreateEstatePage.js

```diff
@@ -0,0 +1,77 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+
+import EstateSelect from './EstateSelect'
+import EditEstate from './EditEstate'
+
+export default class CreateEstatePage extends React.PureComponent {
+  static propTypes = {
+    x: PropTypes.string.isRequired,
+    y: PropTypes.string.isRequired,
+    createEstate: PropTypes.func.isRequired
+  }
+
+  constructor(props) {
+    super(props)
+    const x = parseInt(this.props.x, 10)
+    const y = parseInt(this.props.y, 10)
+    this.state = {
+      estate: {
+        data: {
+          name: '',
+          description: '',
+          parcels: [{ x, y }]
+        }
+      },
+      isSelecting: true
+    }
+  }
+
+  handleSwitch = () => {
+    this.setState({ isSelecting: !this.state.isSelecting })
+  }
+
+  handleChangeParcels = parcels => {
+    const { estate } = this.state
+    this.setState({
+      estate: {
+        ...estate,
+        data: {
+          ...estate.data,
+          parcels
+        }
```

Comment:
I can not see the usage of `...estate`  before `data` based on `initialState`
Maybe this could be enough
```
this.setState({
   estate: {
     data: {
         ...estate.data,
         parcels
      }
   }
})
```

Reply:
Right know it's useless, before it was being use because the estate object was having some more attributes in addition to data object. fixed 14e24c4